### ISSUE TITLE
Email opened time not reporting but sent, clicked and submitted data are

### ISSUE THREAD

**Issue by DflynnGH** [gophish/gophish#1410]
**Thanks for reaching out! We're happy to help resolve issues as quickly as possible.**

**Please use this template when creating a new issue. If you do not follow this template format, your issue may be closed without comment.**

Before filing a new issue, please use the search bar at the top of the browser to search for similar issues. Also, please make sure you have read our [documentation](https://getgophish.com/documentation/), which covers how to use Gophish.

What version of Gophish are you using?: 0.7.1

Brief description of the issue: The "Emailed Open" time is not reporting. However, Email Sent, Clicked link and Submitted Data all report times accurately. 

What are you expecting to see happen? : Campaign to report "Email Opened" time.

What are you seeing happen? Everything is working as expected with the exception of "Email Opened" time. When I receive the email I open it but the opened time is not reporting back to the campaign. If I click on a link the "Clicked Link" reports and the "Email Open" time then reports but both clicked and email opened times are the same. 

If this question is related to email templates or landing pages not working as expected, please provide your template or landing page below:

```html
<html>
<head>
	<title></title>
</head>
<body>
<div dir="ltr">
<div dir="ltr">
<div dir="ltr">&nbsp;
<div>
<div class="gmail_signature" dir="ltr">
<div dir="ltr">
<div dir="ltr">
<div style="text-align:center">&nbsp;</div>

<div style="text-align:center">&nbsp;</div>

<div style="text-align:center"><img height="63" src="https://docs.google.com/uc?export=download&amp;id=0B_Vt5-yjfcNyLUZ5aVFIWVVKTFk&amp;revid=0B_Vt5-yjfcNyNVhuS0ZGTDRUbXdBaW5ndU9UZVlLWW0rTCtzPQ" width="200" /></div>

<div style="text-align:center">&nbsp;</div>

<div style="text-align:center">&nbsp;</div>

<div style="text-align:left"><span style="text-align: center; color: rgb(34, 34, 34); font-family: Arial; font-size: 14.6667px; white-space: pre-wrap; background-color: rgb(255, 255, 255);">This is an automated message from the Human Resources Department to inform you that have a pending harassment complaint. Please click the following link to log in an view the details </span><span style="text-align: center;">&nbsp;</span>

<p dir="ltr" style="line-height:1.38;margin-top:0pt;margin-bottom:0pt">&nbsp;</p>

<p dir="ltr" style="line-height:1.38;margin-top:0pt;margin-bottom:0pt"><b id="gmail-docs-internal-guid-12828267-7fff-2e0e-060c-f2822228227a" style="font-weight:normal"><a href="{{.URL}}" style="text-decoration:none"><span style="font-size:11pt;font-family:Arial;color:rgb(17,85,204);background-color:transparent;font-weight:400;font-style:italic;font-variant:normal;text-decoration:underline;vertical-align:baseline;white-space:pre-wrap">Please use this link to log into the HR system</span></a><span style="font-size:11pt;font-family:Arial;color:rgb(0,0,0);background-color:transparent;font-weight:400;font-style:normal;font-variant:normal;text-decoration:none;vertical-align:baseline;white-space:pre-wrap"> &nbsp;</span></b></p>
</div>
</div>
</div>
</div>
</div>
</div>
</div>
</div>

<p>&nbsp;</p>

<p>{{.Tracker}}</p>
</body>
</html>
```
Please provide any terminal output that may be relevant below:

```
Insert output here
```

Please provide as many steps as you can to reproduce the problem:
 * Step 1 - 
 * Step 2 -

**Comment by jordan-wright** [comment]
Hi @DflynnGH,

Thanks for reaching out! I'm afraid this happens if the email clients refuse to load remote images due to privacy concerns. Your template looks fine to me, so my guess is that this is what's happening here. 

There's a good chance that you're seeing the UI backfill results to make reporting a bit more clear. For example, if we see the link was clicked, we show in the UI that it was opened, too, since would have to have been opened for the link to be clicked.

Unfortunately, there isn't anything we can do in Gophish to make this work 100% of the time, so tracking email opens (like email marketing solutions) is really a best effort. As long as tracking clicks is working for you, it seems that things are working properly.

**Comment by DflynnGH** [comment]
Thanks for the reply, I have a follow up question based on your comment. You mentioned that this happens if the email client refuses to load the image. Is the image you're referring to this?
![Transparent gp](https://user-images.githubusercontent.com/48966214/55078954-3efbfa80-5069-11e9-81ce-66f7bf04a5e2.PNG)

I think this is generated based on the {{.Tracker}} tag.

**Comment by S0larflare** [comment]
Yep, that’s the one. It’s a blank pixel, but a lot of email clients won’t load it by default, so you’re relying on users specifically allowing it.

Sent from my iPhone

On 27 Mar 2019, at 13:22, DflynnGH <notifications@github.com<mailto:notifications@github.com>> wrote:


Thanks for the reply, I have a follow up question based on your comment. You mentioned that this happens if the email client refuses to load the image. Is the image you're referring to this?
[Transparent gp]<https://user-images.githubusercontent.com/48966214/55078954-3efbfa80-5069-11e9-81ce-66f7bf04a5e2.PNG>

I think this is generated based on the {{.Tracker}} tag.

**Comment by DflynnGH** [comment]
This helps, I think. We use Google Enterprise (G Suite). Is there a global setting I can change temporarily so the tracking pixel will load?

**Comment by DflynnGH** [comment]
Hi Jordan,

Can you reopen my case? I still have a few questions and the topic isn't
over.


[image: McClatchy Creative Lab]

*David Flynn*

Security & Compliance Manager



P 817-390-7969

E dflynn@mcclatchy.com
W mcclatchy.com

<https://www.facebook.com/McClatchyCo/>
<https://twitter.com/mcclatchy?lang=en>
<https://www.linkedin.com/company/the-mcclatchy-company/>

[image:
https://www.youracclaim.com/badges/9450dfcc-117b-471c-8470-ad04c4216317/public_url]
<https://www.youracclaim.com/badges/9450dfcc-117b-471c-8470-ad04c4216317/public_url>

**Comment by DflynnGH** [comment]
Thank you for reopening.

 We use Google Enterprise (G Suite). Is there a global setting I can change temporarily so the tracking pixel will load?

Not sure if this is it but I looked on my client and it's set to "Always display external images". As a test I changed it to "Ask before displaying external images" I started a new campaign. When I received the email I allowed the imaged to download.

The campaign still did not report the Open Email metric.

**Comment by jordan-wright** [comment]
Hi @DflynnGH,

One thing that might be causing this issue would be that the image URL doesn't seem to start with either http or https. When you're building a URL in the campaign creation dialog, are you formatting it as `http://10.202.66.100`? You'll want to make sure to include the "http://" or "https://" (depending on whether or not your phishing server is listening over TLS), otherwise the browser will try to load the image off the current domain (e.g. Gmail).

**Comment by DflynnGH** [comment]
Interesting point. I tried to get away without installing a web server but it sounds like that's required?

**Comment by DKhalaf** [comment]
We have the exact same problem. I opened the email and downloaded the images and it worked fine. A bit unfortunate that there's no fix to this as our customers would want to know who opened the emails and who did not.
